### PR TITLE
 only add Display Search Facets to the Homepage

### PR DESCRIPTION
**Pull request by GauravD2t**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes DSpace/DSpace#8426 (if this fixes an issue ticket)

## Description

This PR adds Search filters to the Homepage (only).  Search filters for Community/Collection pages was split off into #2732 

Enabled by default, but can be disabled via this new configuration in config.yml:
```
homePage:
  # Enable or disable the Discover filters on the homepage
  showDiscoverFilters: true/false
```

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t : I gave this a quick test/review today. 

Overall, the code works and the design looks good. But, there's a few concerns I have in reviewing this code.

1. First, with regards to the design.. it looks good except for one thing.  I don't like that the list of "Subcommunities and Collections" is now under a subpath of `/communities/sub-com-col/[:uuid]`.  Here's a screenshot.

subcommunities

This subpath "breaks" our usual URL pattern.  Usually we have `/[object-type]/[uuid]/[sub-path]` (e.g. edit Community is under `/communities/[uuid]/edit` NOT `/communities/edit/[uuid]`).  But, you'd structured it like `/[object-type]/[sub-path]/[uuid]`.  This likely will effect SEO.

I'd also recommend renaming this subpath to be something like `/communities/[uuid]/browse` (as you are browsing it's Collections/Communities).. or something similar, e.g. `/communities/[uuid]/list` or `/communities/[uuid]/tree` or similar.

2. This PR appears to also accidentally include all the changes in #2222.  This makes it slightly harder to review as the code in #2222 has changed while this PR has older copies of that same code.  If you want to keep it that way, that's OK...but we won't be able to review this PR easily until #2222 is completed.

3. Similar to #2222, this PR appears to have no tests (`.spec.ts` files with actual tests).  We require those for DSpace 7.  So, it's not possible for me to move this forward until we can find a way to add tests for the newly added code.

As it is, I think it's likely we'll need to move this to 8.0, as I don't think there's enough time to clean this up for 7.6 and get additional reviews.  It came in too late unfortunately.

File: src/app/shared/search/search-filters/search-filter/search-filter.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filter/search-filter.component.html:2]: It looks like you've accidentally removed accessibility attributes on this `<div>`.  We should add back in the `aria-labelledby` and `role` attributes, unless there's a clear reason why you've removed them.

File: src/app/shared/search/search-filters/search-filter/search-filter.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filter/search-filter.component.html:9]: It's unclear to me why this is now a `<label>` as this now throws accessibility warnings when using the WAVE plugin,  It tells me that this label is "orphaned" because it is not associated with any form element.  Could we just turn this back into a `<span>`?  It's unclear why we've changed this.

File: src/app/shared/search/search-filters/search-filters.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.html:1]: This change is also throwing accessibility warnings in WAVE (again the warning is this is an "orphaned form label").  I don't understand why we need to change this from an `<h3>` to a `<label class="h3">`.  Could you either change this back to an `<h3>` or explain why this change was made?  We need to fix the accessibility here before we can merge.

File: src/app/shared/search/search-filters/search-filters.component.spec.ts
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.spec.ts:64]: Can you explain why this is commented out?  Is it throwing an error?  Can it be safely uncommented?   Usually we do not want to include commented out code unless there's a clear reason for it.

File: src/app/shared/search/search-filters/search-filters.component.ts
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.ts:65]: Small misspelling in this code.  "filterLable" should be spelled "filterLabel".  (This same variable is used elsewhere in the code, so you'll want to copy this change to other classes that use it)

File: src/app/shared/search/search.utils.ts
Comment (tdonohue) [inline src/app/shared/search/search.utils.ts:52]: This doesn't appear to be related to this PR?  This appears to be accidentally copied over from another PR (perhaps #2608).  So, I think this can be removed from this PR.

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5399]: I believe all the keys above this line are from a different PR (#2608). So, I think they can be safely removed from this PR (but please double check that I'm correct)

File: src/assets/i18n/en.json5
Comment (tdonohue) [inline src/assets/i18n/en.json5:5404]: I *think* these two `search.filters` keys are also unused in this PR and can be removed.  But again, double check it please.

File: src/themes/dspace/styles/_global-styles.scss
Comment (tdonohue) [inline src/themes/dspace/styles/_global-styles.scss:21]: This style change might need to be reverted as I think you may need to revert some of the `<label>` tags, unless you see a different way to fix the accessibility warnings from WAVE.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t : Apologies for not getting back to this PR for so long.  Overall, this works and the User Interface looks good.  

However, I've noticed that some of the code changes to the Filters (where you've modified a header to be a `<label>`)  has resulted in new Accessibility warnings on the search filters.  These warnings are easiest to see via [WAVE](https://wave.webaim.org/extension/) on the search page.  
* On our demo site (https://demo.dspace.org/search?query=), the sidebar of Filters has zero WAVE warnings/errors  (there are warnings elsewhere on the page but none in the Filters section).  
* But, with this PR, the Filters section has a new accessibility warning for every filter shown.  I've called out in the code below where these problems seem to be created.

I've also noticed some other minor code cleanup.  Overall, this looks good.  But, we need to fix the accessibility warnings & perform other minor cleanup of the code.  Thanks!

**Comment by tdonohue** [comment]
@GauravD2t : Just as a note, I also wanted to alert you that this PR and #2722 appear to have overlap.  The PRs themselves are very different in purpose.  But in #2722, there is also a new `SubComColSectionComponent` which appears to have the same purpose as your `SubComCollPageComponent`.  Both of these components appear to move the subcommunity/collection list into a new component.  So, only one will be necessary in the codebase (as we don't need two components that do the same thing).

I want to point this out because whichever PR is merged first will cause **code conflict** with the other.   My apologies in advance if this PR is merged second, but it will likely depend heavily on _which PR is fully ready to merge first._

File: src/app/collection-page/collection-page.component.html
Comment (tdonohue) [inline src/app/collection-page/collection-page.component.html:2]: Would it be possible to undo all the changes to this file?  It appears that you may have just realigned the HTML, but it'd be best to revert all changes, otherwise you may cause code conflicts with other PRs.

File: src/app/community-page/community-page.component.html
Comment (tdonohue) [inline src/app/community-page/community-page.component.html:31]: Same here, could you undo the changes to this file too?

File: src/app/community-page/community-page.component.ts
Comment (tdonohue) [inline src/app/community-page/community-page.component.ts:16]: Same here.  This change can be undone

File: src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html
Comment (tdonohue) [inline src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html:6]: I think all the changes to this file can also be undone.  These are specific to Community/Collection pages again

File: src/app/shared/search/search-filters/search-filter/search-filter.component.html
Comment (tdonohue) [reply src/app/shared/search/search-filters/search-filter/search-filter.component.html:2]: As I mentioned last week, you need to undo the changes to this file.  In this area you accidentally removed the `aria-labelledby` and `role` attributes which are required for accessibility.

File: src/app/shared/search/search-filters/search-filter/search-filter.component.html
Comment (tdonohue) [reply src/app/shared/search/search-filters/search-filter/search-filter.component.html:9]: As mentioned last week, it's unclear why this has been turned into a `<label>` as that causes this page to throw accessibility warnings when using the WAVE plugin. It tells me that this label is "orphaned" because it is not associated with any form element. Could we just turn this back into a `<span>`? It's unclear why we've changed this.

File: src/app/shared/search/search-filters/search-filters.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.html:1]: This change is also throwing accessibility warnings in WAVE (again the warning is this is an "orphaned form label"). I don't understand why we need to change this from an `<h3>` to a `<label class="h3">`. Could you either change this back to an `<h3>` or explain why this change was made? We need to fix the accessibility here before we can merge.

File: src/app/shared/search/search-filters/search-filters.component.spec.ts
Comment (tdonohue) [reply src/app/shared/search/search-filters/search-filters.component.spec.ts:64]: This code should be uncommented again.  If it's failing it needs to be fixed, or you need to explain why it's no longer needed.

File: src/app/shared/search/search-filters/search-filters.component.ts
Comment (tdonohue) [reply src/app/shared/search/search-filters/search-filters.component.ts:65]: Again, this has a small misspelling.  `filterLable` should be `filterLabel`

File: src/app/shared/search/search.utils.ts
Comment (tdonohue) [reply src/app/shared/search/search.utils.ts:52]: Again, this code change doesn't appear to be related to this PR? This appears to be accidentally copied over from another PR (perhaps #2608). So, I think this can be removed from this PR.

File: src/themes/dspace/styles/_global-styles.scss
Comment (tdonohue) [inline src/themes/dspace/styles/_global-styles.scss:20]: Why have you removed the default style for `<h4>`?  This change likely should be undone unless there's a clear reason why this was done.

File: src/app/shared/sidebar/sidebar-dropdown.component.html
Comment (tdonohue) [inline src/app/shared/sidebar/sidebar-dropdown.component.html:2]: I don't understand here why we had to change an `<h4>` into a `<label>`?  Could you explain the purpose of this change?

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t : I gave this an updated code review today.  It appears you removed some of the Community/Collection homepage changes, but **not all the changes**.

I've also noted places inline below where my prior feedback has not yet been addressed.  Please remember to either address all feedback or respond back by answering the questions.  Thanks!

File: src/app/shared/search/search.utils.ts
Comment (GauravD2t) [reply src/app/shared/search/search.utils.ts]: remove here

File: src/app/shared/search/search-filters/search-filters.component.spec.ts
Comment (GauravD2t) [reply src/app/shared/search/search-filters/search-filters.component.spec.ts]: SearchFiltersComponent
    when the getSearchLink method is called
      × should call getSearchLink on the searchService


the error generated while run yarn run test:headless ..

File: src/app/shared/search/search-filters/search-filters.component.html
Comment (GauravD2t) [reply src/app/shared/search/search-filters/search-filters.component.html]: 1 accessibility violation was detected
[
  {
    id: 'heading-order',
    impact: 'moderate',
    description: 'Ensures the order of headings is semantically correct',
    helpUrl: 'https://dequeuniversity.com/rules/axe/4.7/heading-order?application=axeAPI',
    nodes: 1,
    html: [ '<h3 _ngcontent-dspace-angular-c275="">Discover</h3>' ]
  }
]

File: src/app/shared/sidebar/sidebar-dropdown.component.html
Comment (GauravD2t) [reply src/app/shared/sidebar/sidebar-dropdown.component.html]: 1 accessibility violation was detected

File: src/app/collection-page/collection-page.component.html
Comment (tdonohue) [reply src/app/collection-page/collection-page.component.html:2]: @GauravD2t : The changes to the `collection-page.component.html` are still not undone. They still are visible in the files changed tab in GitHub: https://github.com/DSpace/dspace-angular/pull/2275/files

File: src/app/community-page/community-page.component.html
Comment (tdonohue) [reply src/app/community-page/community-page.component.html:31]: @GauravD2t : The changes to the `community-page.component.html` are still not undone. They still are visible in the files changed tab in GitHub: https://github.com/DSpace/dspace-angular/pull/2275/files

File: src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html
Comment (tdonohue) [reply src/app/shared/comcol/comcol-page-browse-by/comcol-page-browse-by.component.html:6]: @GauravD2t : The changes to the `comcol-page-browse-by.component.html` are still not undone. They still are visible in the files changed tab in GitHub: https://github.com/DSpace/dspace-angular/pull/2275/files

File: src/themes/dspace/styles/_global-styles.scss
Comment (tdonohue) [inline src/themes/dspace/styles/_global-styles.scss:20]: @GauravD2t : Can you remove the space added here?  We want to avoid reformatting code unless it's necessary as it makes the PR very messy to review.

File: src/app/shared/sidebar/sidebar-dropdown.component.html
Comment (tdonohue) [inline src/app/shared/sidebar/sidebar-dropdown.component.html:2]: @GauravD2t : Can you **add back in** the space you removed from here? The alignment is now messed up in this file as the `<h4>` should be under the `<div>`. We want to avoid reformatting code unless it's necessary as it makes the PR very messy to review.

File: src/themes/dspace/styles/_global-styles.scss
Comment (GauravD2t) [reply src/themes/dspace/styles/_global-styles.scss]: remove the space befor <h4> tag

File: src/app/search-navbar/search-navbar.component.ts
Comment (artlowel) [inline src/app/search-navbar/search-navbar.component.ts:70]: Currently, if you have a search page open with a number of filters (e.g. https://sandbox.dspace.org/search?spc.page=1&f.author=Simmons,%20Cameron,equals ) and you enter a query from the header, you won't lose those filters.

Removing 'merge' here will ensure that you will lose any preexisting filters. I'm not sure whether or not we want this, but it should be a conscious decision.

I'm also not sure why the change was needed for this PR

**Review by artlowel — DISMISSED** [review]
Thanks @GauravD2t!

It works as advertised, however I'd make this feature configurable and disable it by default

File: src/app/shared/search/search-filters/search-filter/search-filter.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filter/search-filter.component.html:2]: You should be able to also undo all the changes in this file.  It appears you just removed the spaces from this file which made the code aligned better.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@GauravD2t :  I took another look at this today, and it appears that the code cleanup that I asked for still hasn't been done.  It's still visible in GitHub.

We need you to revert the changes to the Collection/Community homepage files in order to move this forward.  I cannot merge this until that is done.  Thanks.

**Review by tdonohue — CHANGES_REQUESTED** [review]
Hi @GauravD2t :  I gave this another look/test today.  It's working, but some of the prior feedback has still not been addressed and tests are failing.  Overall, this is looking much better and seems nearly ready... we just need to have an option to disable it.

1. In [his review](https://github.com/DSpace/dspace-angular/pull/2275#pullrequestreview-1870226140), @artlowel  noted that we should provide an option to disable this feature as not all sites will want these facets on the homepage.  I had also mentioned that in the original design discussion in [this comment on the original ticket.](https://github.com/DSpace/DSpace/issues/8426#issuecomment-1329579857)  I've copied my comment below:
    * "I also want to note that there are potential performance concerns with putting too much content on the homepage. At the very least, this feature needs to be possible to disable if it exists on the homepage."
    * To fix add a disable option, I'd recommend adding a new config similar to this:
       ```
       # This section already exists in `config.example.yml`
       homePage:
         showDiscoverFilters: true/false
       ```
2. Minor issue: I've noted the lines below where tests are failing and how to fix them.
3. Minor issue: There's a misalignment of code issue that I noted previously in `search-filter.component.html`.  It looks like you can simply undo all changes to that file...as the only changes are to accidentally misalign the code (making the code harder to read)

Overall, this is looking good & I thank you for your hard work here.

File: src/app/shared/search/search-filters/search-filters.component.html
Comment (tdonohue) [inline src/app/shared/search/search-filters/search-filters.component.html:1]: The current e2e tests are failing because of an accessibility issue here.  The error is: `'Ensures the order of headings is semantically correct',`

The reason for the error is that the accessibility scanner is expecting this to be an `<h2>` on the homepage.  But it **must** be an `<h3>` everywhere else.

I think we need to look into whether there's a way to make this an `<h2>` ONLY on the homepage, while it's kept as an `<h3>` everywhere else.

**Review by tdonohue — COMMENTED** [review]
@GauravD2t : Thanks for your hard work on this!  I'm nearly a +1 as I can verify this is working properly.  However, currently our automatic accessibility tests are failing on the homepage because the "Discover" header is an `<h3>` when it expects an `<h2>` (as there's no `<h2>` heading above that `<h3>`).    I think we should try to find a way to fix this within this PR if possible.  (If somehow this is too difficult, we could also consider temporarily disabling the failing accessibility test -- but I'm hopeful we won't need to do that.)

More information on the accessibility error inline below.

File: src/app/shared/search/search-filters/search-filters.component.html
Comment (GauravD2t) [reply src/app/shared/search/search-filters/search-filters.component.html]: <h3 *ngIf="inPlaceSearch">{{filterLabel+'.filters.head' | translate}}</h3>
<h2 *ngIf="!inPlaceSearch">{{filterLabel+'.filters.head' | translate}}</h2>

**Review by artlowel — COMMENTED** [review]
Thanks @GauravD2t 

The new configuration option works for me, but I would still disable the feature by default. I'm not a fan of how it makes the homepage take up the full width of the window either

**Comment by tdonohue** [comment]
@GauravD2t : Could you change the configuration so that this feature is disabled by default as requested by @artlowel [above](https://github.com/DSpace/dspace-angular/pull/2275#pullrequestreview-1895533907)?  Thanks.  I'll try to give this another test today to make sure my feedback has been addressed.

File: config/config.example.yml
Comment (tdonohue) [inline config/config.example.yml:276]: Please change default to `false`

File: src/config/default-app-config.ts
Comment (tdonohue) [inline src/config/default-app-config.ts:278]: Please change default to `false`

File: src/config/homepage-config.interface.ts
Comment (tdonohue) [inline src/config/homepage-config.interface.ts:25]: I think this should say:
```
showDiscoverFilters: boolean;
```

That's how this appears in other `config.interface.ts` files, e.g. [`markdown-config.interface.ts](https://github.com/DSpace/dspace-angular/blob/main/src/config/markdown-config.interface.ts#L14)`

**Review by tdonohue — COMMENTED** [review]
@GauravD2t : Thanks for the updates.  I've given this a test today & all my prior feedback has been addressed.  I think the **final step** is to disable this by default (as @artlowel requested above).  I've added some notes inline on where those minor changes are necessary.

After these changes are made, I'll give it a final look & then squash merge it (so that it goes in as one commit instead of 70+).  Thanks!

**Pull request merged**
